apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -230,20 +269,26 @@ def _maybe_eager_iterator(
 
     def _table_reader(self) -> Iterator[BlockwiseTableReadIterResult]:
         """Private. Blockwise table reader. Helper function for sub-class use"""
-        kwargs: Dict[str, object] = {"result_order": self.sr.result_order}
         for coord_chunk in _coords_strider(
             self.coords[self.major_axis],
-            self.sr.shape[self.major_axis],
+            self.shape[self.major_axis],
             self.size[0],
         ):
-            self.sr.reset(**kwargs)
             step_coords = list(self.coords)
             step_coords[self.major_axis] = coord_chunk
-            _util._set_coords(self.sr, step_coords)
 
             joinids = list(self.joinids)
             joinids[self.major_axis] = pa.array(coord_chunk)
-            yield pa.concat_tables(_arrow_table_reader(self.sr)), tuple(joinids)
+            yield pa.concat_tables(
+                ArrowTableRead(
+                    array=self.array,
+                    coords=step_coords,
+                    column_names=[],
```

Comment:
I'm not understanding why this is `[]`

Reply:
This means we select all columns for reads. I will write an inline comment for clarification.